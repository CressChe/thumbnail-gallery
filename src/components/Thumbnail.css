.thumbnail-gallery {
    background: white;
    height: 500px;
    width: 550px;
    margin: 40px auto;
    display: flex;
    margin-left: 25%;
}
.thumbnails-grid {
   /* height: 35%;
    width: 100%;*/
    background: white;
    display: flex;
    flex-wrap: wrap;
}
.title-style {
    font-size: medium;
    text-align: left;
    background-color: white;
}

.active-window {
    /* height: 65%;
    width: 100%; */
    background: white;
}

.image-window {
    /* width: 100%;
    height: 100%; */
}

.thumbnail-styles {
    height: 50%;
    width: 24.63%;
    border: white;
    border-width: thin;
    border-style: solid;
}
.selected-style {
    /* width: 90%;
    height: 90%; */
    border: red;
    border-style: solid;
    border-width: thick;
  }

.image-styles {
    /* width: 100%;
    height: 100%; */
}
.text-style {
    font-size: small;
    text-align: left;
    width: 100%;
}