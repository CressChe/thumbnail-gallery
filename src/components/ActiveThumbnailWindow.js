import './Thumbnail.css';

export const ActiveThumbnailWindow = ({ activeId }) => {

    return (
        <div class='active-window'>
            <img class='image-window' alt={activeId} src={`https://picsum.photos/id/${activeId}/200/300.jpg`} data-testid="image" />
        </div>
    )
}