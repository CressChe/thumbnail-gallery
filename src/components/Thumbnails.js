import './Thumbnail.css';

export function Thumbnails({ id, setActiveId, activeId }) {
    const style = id === activeId ? 'selected-style' : 'image-styles';
    
    return (
        <div class='thumbnail-styles' >
            <img class={style} alt={id} src={`https://picsum.photos/id/${id}/100/90.jpg`} data-testid={`thumbnail-image-${id}`} onClick={() => setActiveId(id)} />
        </div>
    )
}
