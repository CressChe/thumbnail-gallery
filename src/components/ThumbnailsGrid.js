import { Thumbnails } from "./Thumbnails";
import './Thumbnail.css';

export function ThumbnailsGrid({ selectedIds, setActiveId, activeId }) {

    return (
        <div class='thumbnails-grid'>
            <h3 class="text-style">Select your photo</h3>
            {   
                selectedIds.map((id) => {
                    return (
                        <Thumbnails key={id} id={id} setActiveId={setActiveId} activeId={activeId} />
                    )
                })
            }
        </div>
    )
}

