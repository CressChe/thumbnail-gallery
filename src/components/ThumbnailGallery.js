import { ActiveThumbnailWindow } from "./ActiveThumbnailWindow";
import { ThumbnailsGrid } from "./ThumbnailsGrid";
import { useState } from 'react';
import './Thumbnail.css';

export function ThumbnailGallery() {
    const selectedIds = [1015, 1016, 1019, 1022, 1039, 1043, 1057, 11];

    const [activeId, setActiveId] = useState(selectedIds[0])

    return (
        <div class="thumbnail-gallery">
            <div>
            <h1 class="title-style"> React Photo Viewer</h1>
                <ActiveThumbnailWindow activeId={activeId} />
                <ThumbnailsGrid selectedIds={selectedIds} setActiveId={setActiveId} activeId={activeId}/>
            </div>
        </div>
    );
}
