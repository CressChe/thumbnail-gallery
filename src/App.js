import {ThumbnailGallery} from './components/ThumbnailGallery'


function App() {
  return (
    <div className="App">
      <header style={header}>
        <ThumbnailGallery />
      </header>
    </div>
  );
}
const header = { 
  width: '50%',
  textAlign:'center',
}

export default App
