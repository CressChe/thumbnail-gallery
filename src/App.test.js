import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import App from './App';
import {ActiveThumbnailWindow} from './components/ActiveThumbnailWindow';
import {Thumbnails} from './components/Thumbnails';
import renderer from 'react-test-renderer';
import {ThumbnailGallery} from './components/ThumbnailGallery';


test('renders select your photo', () => {
  render(<App />);
  const linkElement = screen.getByText(/Select your photo/i);
  expect(linkElement).toBeInTheDocument();
});

test('Checks active thumbnail window renders images with active Id', () => {
  render(<ActiveThumbnailWindow activeId={1015}/>);

  const img = screen.getByTestId("image");

  expect(img).toBeInTheDocument();
  expect(img.src).toBe("https://picsum.photos/id/1015/200/300.jpg")
});

test('Checks thumbnail window renders images with id', () => {
  render(<Thumbnails id={1022}/>);

  const img = screen.getByTestId("thumbnail-image-1022");

  expect(img).toBeInTheDocument();
  expect(img.src).toBe("https://picsum.photos/id/1022/100/90.jpg")
});

test('should match snapshot of Active Window view', () => {
  const activeWindow = renderer.create(<ActiveThumbnailWindow/>).toJSON();

  expect(activeWindow).toMatchSnapshot();
});

test('should match snapshot of Active Window when activeId is set', () => {
  const activeWindow = renderer.create(<ActiveThumbnailWindow activeId={1022}/>).toJSON();

  expect(activeWindow).toMatchSnapshot();
})

test('should change active window image when a thumbnail image is clicked', async () => {
  render(<ThumbnailGallery activeId={1015}/>);
  const thumbnail = screen.getByTestId("thumbnail-image-11");

  userEvent.click(thumbnail);

  const image = screen.getByTestId("image");
  expect(image.src).toBe("https://picsum.photos/id/11/200/300.jpg")
})