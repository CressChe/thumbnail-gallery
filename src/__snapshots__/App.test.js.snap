// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of Active Window view 1`] = `
<div
  class="active-window"
>
  <img
    class="image-window"
    data-testid="image"
    src="https://picsum.photos/id/undefined/200/300.jpg"
  />
</div>
`;

exports[`should match snapshot of Active Window when activeId is set 1`] = `
<div
  class="active-window"
>
  <img
    alt={1022}
    class="image-window"
    data-testid="image"
    src="https://picsum.photos/id/1022/200/300.jpg"
  />
</div>
`;
